File: python/201/loop_python_ex/factor.py
import math

n = int(raw_input("What number do you want factored? "))
f = []
sq = int(math.floor(n**0.5) + 1)
for x in range(1,sq):
    if n % x == 0:
        f.append(x)
        if f.count(n/x) == 0:
            f.append(n/x)
f.sort()
print "This is a list of factors for your number: " + str(f)

File: javascript/101/factor.js
function factors(num) {
  var f = [];
  for (var i=1; i<=Math.sqrt(num); i++) {
    if (num % i === 0) {
      if (num / i !== i) {
        f.push(num/i);
      }
      f.push(i);
    }
  }
  f.sort(function(a,b){return a-b;});
  return f;
}

console.log(factors(16));
